src/main/java/com/amazon/ion/impl/IonReaderTextSystemX.java

```diff
@@ -1371,6 +1387,164 @@ public boolean hasNext()
+    public void prepareTranscodeTo(@NotNull MacroAwareIonWriter writer) {
+        macroAwareTranscoder = writer;
+    }
+
+    @Override
+    public boolean transcodeNext() throws IOException {
+        // TODO consider improving the readability of this method and its binary counterpart: https://github.com/amazon-ion/ion-java/issues/1004
+        if (macroAwareTranscoder == null) {
+            throw new IllegalArgumentException("prepareTranscodeTo must be called before transcodeNext.");
+        }
+        boolean isSystemValue = false;
+        while (true) {
+            if (isEvaluatingEExpression) {
+                if (evaluateNext()) {
+                    if (isSystemValue) {
+                        continue;
+                    }
+                    return !_eof;
+                }
+            } else {
+                nextRaw();
+            }
+            isSystemValue = false;
+            if (_value_type != null && getDepth() == 0) {
+                if (IonType.SYMBOL == getType() && handlePossibleIonVersionMarker()) {
+                    // Which IVM to write is inherent to the writer implementation.
+                    // We don't have a single implementation that writes both formats.
+                    macroAwareTranscoder.startEncodingSegmentWithIonVersionMarker();
+                    isSystemValue = true;
+                    continue;
+                }
+                if (minorVersion > 0 && isPositionedOnEncodingDirective()) {
+                    boolean isEncodingDirectiveFromEExpression = isEvaluatingEExpression;
+                    readEncodingDirective();
+                    macroAwareTranscoder.startEncodingSegmentWithEncodingDirective(
+                        encodingDirectiveReader.getNewMacros(),
+                        encodingDirectiveReader.isMacroTableAppend(),
+                        encodingDirectiveReader.getNewSymbols(),
+                        encodingDirectiveReader.isSymbolTableAppend(),
+                        isEncodingDirectiveFromEExpression
+                    );
+                    isSystemValue = true;
+                    continue;
+                }
+            }
+            if (_container_is_e_expression) {
+                expressionArgsReader.beginEvaluatingMacroInvocation(macroEvaluator);
+                macroEvaluatorIonReader.transcodeArgumentsTo(macroAwareTranscoder);
+                isEvaluatingEExpression = true;
+                continue;
+            } else if (isEvaluatingEExpression) {
+                // This is an e-expression that yields user values. Its arguments have already been transcoded.
+                continue;
+            }
+            transcodeValueLiteral();
+            return !_eof;
+        }
+    }
+
+    /**
+     * @return true if the reader is positioned on an Ion 1.0 symbol table; otherwise, false. Note: the caller must
+     *  ensure this is called only at the top level.
+     */
+    boolean isPositionedOnSymbolTable() {
```

Comment:
The green from here through `handlePossibleIonVersionMarker` was moved from the user-level reader.